src/main.zig

```diff
@@ -1711,47 +1712,55 @@ fn buildOutputType(
             if (runtime_args_start) |i| {
                 try argv.appendSlice(all_args[i..]);
             }
-            // TODO On operating systems that support it, do an execve here rather than child process,
-            // when watch=false and arg_mode == .run
-            const child = try std.ChildProcess.init(argv.items, gpa);
-            defer child.deinit();
+            if (std.builtin.os.tag == .linux and arg_mode == .run and watch == false) { // TODO does other posix os support execve in zig stdlib?
+                var env_vars = try process.getEnvMap(gpa);
+                defer env_vars.deinit();
+                const term = os.execvpe(gpa, argv.items, &env_vars);
```

Comment:
```suggestion
                try os.execvpe(gpa, argv.items, &env_vars);
```

You're discarding the error here.

Reply:
I had origionally tried that but I found that execvpe doesn't return an error union
```
./src/main.zig:1718:31: error: expected error union type, found 'std.os.error:1396:16'
                try os.execvpe(gpa, argv.items, &env_vars);
                              ^
```
What is the best thing to do here?

Reply:
You could simply return the error:
```suggestion
                return os.execvpe(gpa, argv.items, &env_vars);
```

Reply:
That makes sense! Thanks.

Reply:
Alright, I think that's a workaround to avoid hitting #3461. You can either `return` or do something a bit nicer like:

```zig
const err = os.execvpe(gpa, argv.items, &env_vars);
fatal("shit hit the fan : {}", .{@errorName(err)});
```